src/main/java/core/basesyntax/interfaces/FruitOperationImpl.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.interfaces;
+
+import core.basesyntax.FruitRecord;
+import java.util.Map;
+
+public class FruitOperationImpl implements FruitOperation {
+
+    @Override
+    public void process(FruitRecord record, Map<String, Integer> stock) {
+
+        stock.putIfAbsent(record.getFruit(), 0);
+        int current = stock.get(record.getFruit());
+
+        switch (record.getOperation()) {
+            case "b":
+                stock.put(record.getFruit(), record.getQuantity());
+                break;
+            case "s":
+            case "r":
+                stock.put(record.getFruit(), current + record.getQuantity());
+                break;
+            case "p":
+                stock.put(record.getFruit(), current - record.getQuantity());
+                break;
+            default:
+                System.out.println("There is a problem with data!");
```

Lines 25-26 (default branch): Printing an error to stdout violates the checklist guidance: "Don't ignore exceptions in try-catch blocks" and the associated recommendation: "Don't leave `e.printStackTrace()` or `System.out.println(e.getMessage())` in the catch blocks. Let's rethrow a RuntimeException with an **informative** message and exception object." Replace `System.out.println` with throwing an informative RuntimeException that includes the unknown operation code and the record context.